### PR TITLE
remove numpy constraint

### LINKED ISSUE #99: scipy with MKL
It seems like the current build doesn't work with conda's MKL is this expected?

tagging @isuruf

```
conda create -n test_scipy scipy "libblas=*=*mkl"
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.6.14

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: /home/tinkerer/.conda/envs/test_scipy

  added / updated specs:
    - libblas[build=*mkl]
    - scipy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    numpy-1.16.3               |   py37he5ce36f_0         4.3 MB  conda-forge

The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h14c3975_1002   conda-forge
    ca-certificates: 2019.3.9-hecc5488_0   conda-forge
    certifi:         2019.3.9-py37_0       conda-forge
    intel-openmp:    2019.3-199            defaults
    libblas:         3.8.0-10_mkl          conda-forge
    libcblas:        3.8.0-10_mkl          conda-forge
    libffi:          3.2.1-he1b5a44_1006   conda-forge
    libgcc-ng:       8.2.0-hdf63c60_1      defaults
    libgfortran-ng:  7.3.0-hdf63c60_0      defaults
    liblapack:       3.8.0-10_mkl          conda-forge
    libstdcxx-ng:    8.2.0-hdf63c60_1      defaults
    mkl:             2019.3-199            defaults
    ncurses:         6.1-hf484d3e_1002     conda-forge
    numpy:           1.16.3-py37he5ce36f_0 conda-forge
    openssl:         1.1.1b-h14c3975_1     conda-forge
    pip:             19.1-py37_0           conda-forge
    python:          3.7.3-h5b0a415_0      conda-forge
    readline:        7.0-hf8c457e_1001     conda-forge
    scipy:           1.3.0-py37hab63836_0  conda-forge
    setuptools:      41.0.1-py37_0         conda-forge
    sqlite:          3.28.0-h8b20d00_0     conda-forge
    tk:              8.6.9-h84994c4_1001   conda-forge
    wheel:           0.33.4-py37_0         conda-forge
    xz:              5.2.4-h14c3975_1001   conda-forge
    zlib:            1.2.11-h14c3975_1004  conda-forge

Proceed ([y]/n)?


Downloading and Extracting Packages
numpy-1.16.3         | 4.3 MB    | ####################################################################################################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate test_scipy
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(py37) x86_64-conda_cos6-linux-gnu ➜  ~  conda activate test_scipy
(test_scipy) x86_64-conda_cos6-linux-gnu ➜  ~  ipython
import sciptPython 3.7.1 | packaged by conda-forge | (default, Feb 18 2019, 01:42:00)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import scipy.linalg
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-810a6e7acbc7> in <module>
----> 1 import scipy.linalg

~/.conda/envs/py37/lib/python3.7/site-packages/scipy/linalg/__init__.py in <module>
    193 from .linalg_version import linalg_version as __version__
    194
--> 195 from .misc import *
    196 from .basic import *
    197 from .decomp import *

~/.conda/envs/py37/lib/python3.7/site-packages/scipy/linalg/misc.py in <module>
      3 import numpy as np
      4 from numpy.linalg import LinAlgError
----> 5 from .blas import get_blas_funcs
      6 from .lapack import get_lapack_funcs
      7

~/.conda/envs/py37/lib/python3.7/site-packages/scipy/linalg/blas.py in <module>
    212 import numpy as _np
    213
--> 214 from scipy.linalg import _fblas
    215 try:
    216     from scipy.linalg import _cblas

ImportError: libopenblas.so.0: cannot open shared object file: No such file or directory
```

### PR DESCRIPTION
**Pull request by basnijholt**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #99

### CODE REVIEW

**Comment by basnijholt** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by basnijholt** [comment]
The tests seem to pass, merge?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,8 +28,7 @@ requirements:
     - python
     - setuptools
     - cython
-    - numpy 1.13.3  # [py<37]
-    - numpy 1.14.5  # [py>=37]
+    - numpy
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Forgot to bump the build number. No big deal b/c I moved the previous `0` to broken and the hash is different, but it is prefer to always bump for new packages with the same version.

Comment (basnijholt) [reply recipe/meta.yaml]: I did do it in this commit https://github.com/conda-forge/scipy-feedstock/pull/100/commits/f560d4d85174b480d4495ca5995bcc33a967bbc8. Must have got lost somehow.

Comment (isuruf) [reply recipe/meta.yaml]: Sorry about that. I was editing in GitHub and it got lost

Comment (ocefpaf) [reply recipe/meta.yaml]: Not a problem. I do that all the time when editing online :smile:

**Pull request merged**